RFCXXX-Native-Json-Adapters.md

```diff
@@ -0,0 +1,258 @@
+structured data via means of a generalized converter.
+This would require some sort of registration/configuration to designate the native application and call the
+converter to automatically construct objects from tabular data could be called.
+This would enable the user to set it once and later sessions would be able to use it.
+
+#### PSDefaultNativeParameterValues
+
+Additionally, we can extend the _concept_ of the `$PSDefaultParameterValue` variable to improve the user experience by automatically
+adding parameters and their values to the invocation of the native command.
+This would enable our customers to type:
+
+```powershell
+PS> docker image ls
+```
+
+rather than
+
+```powershell
+PS> docker image ls --format='{{json .}}'
+```
+
+Further, it would be better to extend the `$PSDefaultParameterValue` behavior because parameter binding is done differently;
+some parameters must be in a specific location in the command line.
+For example, there are certain parameters which apply to the executable and not the subcommand:
+
+```powershell
+PS> docker --debug run --rm -it mcr.microsoft.com/powershell:latest 
+```
+
+The use of `--debug` has a required position, it _must_ provided before the `run` subcommand.
+
+An addition complication is that depending on the application, parameters should _not_ be added based on the earlier parts of the command.
+For example, you may want to provide different formatting for `git log` vs `git branch`, so you must have more information to determine the proper behavior.
+This means that rather than just capturing the parameter name and default value, we will need to be able to run arbitrary code (ala scriptblock)
+which includes the command line AST so the proper parameter values may be created.
+
+However, with an approach which provides for this complexity, we can continue to simplify the experience for our users to from:
+
+```powershell
+PS> docker --debug run --rm -it mcr.microsoft.com/powershell
+```
+
+to
+
+```powershell
+PS> docker run mcr.microsoft.com/powershell
+```
+
+or
+
+```powershell
+PS> Docker image ls --format='{{json .}}' | ConvertFrom-Json | Where-Object {$_.Repository -match "mcr"}| Format-Table Id,Repository,Tag,Size,CreatedAt
+
+ID           Repository                             Tag                     Size   CreatedAt
+--           ----------                             ---                     ----   ---------
+4e9617ada198 mcr.microsoft.com/powershell           preview-7.3-mariner-2.0 235MB  2022-09-01 11:48:17 -0700 PDT
+. . .
+```
+
+to this:
+
+```powershell
+PS> docker image ls | ? Repository -match "mcr"
+
+ID           Repository                             Tag                     Size   CreatedAt
+--           ----------                             ---                     ----   ---------
+4e9617ada198 mcr.microsoft.com/powershell           preview-7.3-mariner-2.0 235MB  2022-09-01 11:48:17 -0700 PDT
+. . .
+```
+
+### Follow-on Work
```

Comment:
Define tag for publishing modules to PSGallery that includes json adapters

Reply:
Would there be anything stopping someone from making an adapter that is just a keylogger or something that steals the users data?